### DISCUSSION TITLE
How to listen filter changed event in a Ressource

### DISCUSSION THREAD

**Discussion by lepresk** [filamentphp/filament#6612] [Help] [answered]
Hello everyone, I'm new to Filament and I'm stuck.

I've created a `ProductResource` resource and a `ProductOverview` widget that is displayed in the header widgets of the `ListProducts` page. `ProductOverview` is a `StatsOverviewWidget` that displays some information such as the number of products, total stock value, etc.

I would like to retrieve the query containing the filters applied to the table when the user changes the filters. I need to do this in the `getCards` method in order to perform summary operations (sum, count, avg) and update my `ProductOverview`.

**Comment by jackleblack-kpt** [comment]
Up

**Comment by danharrin** [comment]
You can't retrieve the query, as you cant access the Page object from the Widget object. The best you would be able to do is send the array of filter data to the widget and then scope the query with that.

**Reply by lepresk** [reply]
How can procced to retrieve filters applied as an array, I know with Livewire I can use the event system to dispatch a filterChangedEvent for example but how do I listen when filters applied to the table changed ? Can you provide a sample code ?

**Reply by danharrin** [reply]
It would be something similar to this, with the `updated()` method: https://filamentphp.com/blog/how-to-refresh-widgets-when-table-actions-are-fired

**Reply by lepresk** [reply]
Thanks, I've read the article and it's interesting in a way, but it doesn't cover the problem I'm having, which is recovering the filters currently active on the table.

**Reply by danharrin** [reply]
You need to adapt it a bit. You need to emit an event when the filters change (there is an example of this near the end of the article), listen for it on the widget, and then interpret the filters array

**Reply by lepresk** [reply]
I can emit an event when the filters change, but I cannot retrieve the applied filters. That's where the problem lies.

**Reply by danharrin** [reply]
What have you tried

**Reply by lepresk** [reply]
The `updated` method as shown below is triggered when a filter changes, but only one filter at a time. The worst part is that it is not called at all when I remove an applied filter or when I reset all applied filters.

```php
   public function updated($name, $value): void
    {
        if (Str::of($name)->contains('tableFilters')) {
            dd($value);
        }
    }
```

**Reply by danharrin** [reply, marked as answer]
There are different methods that get called when the filters are reset or removed, have a look in the codebase and see if you can find them

In the `updated`, `dd($this->tableFilters)` gives you the array

**Reply by lepresk** [reply]
Great, it's works like a charm
